lib/parse.rb

```diff
@@ -4,38 +4,70 @@ class Parse
   def initialize(path)
     @path = path
     @lines = []
+    @game_info = {}
   end
 
   def fopen
-    if File.exists?(@path)
+    if File.exist?(@path)
       file = File.open(@path)
       begin
-        @lines = file.readlines.map(&:chomp)
+        fread(file)
       rescue
-        return nil
+        nil
       ensure
         file.close
       end
     end
   end
 
-  def get_first_line
-    return @lines[0]
+  def first_line
+    @lines[0]
   end
 
   def output
-    my_object = {
-      "#{@path}": {
-        lines: count_lines()
-      } 
+    @game_info = {
+      @path => {
+        lines: count_lines(),
+        players: [],
```

Comment:
Sugestão, ao invés de escrever um método que altera diretamente o estado da variável de instância. Faça com que o método retorna a lista de players

```suggestion
        players: obtain_player_name(),
```